test/nightly/azureResourceMongoDB.test.ts

```diff
@@ -38,43 +37,22 @@ suite('MongoDB action', async function (this: ISuiteCallbackContext): Promise<vo
         assert.ok(createAccount);
     });
 
-    test('Create MongoDB Database', async () => {
+    test('Create Mongo Database', async () => {
         const databaseName: string = randomUtils.getRandomHexString(12);
         const collectionName: string = randomUtils.getRandomHexString(12);
         const testInputs: string[] = [testAccount.getSubscriptionContext().subscriptionDisplayName, `${accountName} (MongoDB)`, databaseName, collectionName];
         await testUserInput.runWithInputs(testInputs, async () => {
             await vscode.commands.executeCommand('cosmosDB.createMongoDatabase');
         });
-        const response: string = await getMongoDBDatabase(await getUrl(databaseName));
-        assert.equal(JSON.parse(response).properties.id, databaseName);
-    });
-
-    async function getUrl(Path: string): Promise<string> {
-        let baseUrl: string = testAccount.getSubscriptionContext().environment.resourceManagerEndpointUrl;
-        return baseUrl + (baseUrl.endsWith('/') ? '' : '/') + `subscriptions/${client.subscriptionId}/resourceGroups/${resourceGroupName}/providers/Microsoft.DocumentDB/databaseAccounts/${accountName}/apis/mongodb/databases/${Path}?api-version=2015-04-08`;
-    }
-});
-
-async function signRequest(req: WebResource, cred: ServiceClientCredentials): Promise<void> {
-    await new Promise((resolve: () => void, reject: (err: Error) => void): void => {
-        cred.signRequest(req, (err: Error | undefined) => {
-            if (err) {
-                reject(err);
-            } else {
-                resolve();
-            }
+        await vscode.env.clipboard.writeText('');
+        await testUserInput.runWithInputs([`${accountName} (MongoDB)`], async () => {
+            await vscode.commands.executeCommand('cosmosDB.copyConnectionString');
         });
+        const connectionString: string = await vscode.env.clipboard.readText();
+        const mongoClient: MongoClient | undefined = await connectToMongoClient(connectionString, appendExtensionUserAgent())
+        let listDatabases: { databases: IDatabaseInfo[] } = await mongoClient.db(accountName).admin().listDatabases();
+        const MongoDatabase: IDatabaseInfo[] = listDatabases.databases.filter((database: IDatabaseInfo) => (database.name == databaseName));
+        assert.ok(MongoDatabase[0].name == databaseName, `Mongo Database should be ${databaseName} rather than ${MongoDatabase[0].name}.`);
```

Comment:
If the mongoDatabase is not found (the filter fails to return anything) then this will fail as you cannot read `undefined.name`.  I think you could do `listDatabases.databases.find()` or .`some()` and see if it returns a value.